### ISSUE TITLE
1.705 buggy as hell & other multiple nuisances of YPD 

### ISSUE THREAD

**Issue by Droyk** [shaked6540/YoutubePlaylistDownloader#15]
Hey Dev, So today I updated my YPD to 1.705 thinking all the problems that I had in 1.72 & 1.704 must have been fixed but I was dead wrong. anyway, after the update I put a link to download a video & it successfully manages to download the video however it didn't download the subtitles of the video and for some weird reasons converting part of the process just doesn't end
![record_190120_090802](https://user-images.githubusercontent.com/15848350/51435010-cf961300-1c93-11e9-9424-f77cb9ab09d3.gif)

- Download counter is still broken. ATM I don't have a good working internet connection. I Can only get 10 Mb/s at best. here are the results I got, first of all here is my Download speed in [MiB/s](https://user-images.githubusercontent.com/15848350/51422215-db6cd100-1bd0-11e9-93ca-7665bdebe9b9.png) & here it is in [Mb/s](https://user-images.githubusercontent.com/15848350/51422216-ddcf2b00-1bd0-11e9-901f-e021f3989790.png) so only a 10Mb/s at best & now here is the result of your app download counter
![record_190119_044543](https://user-images.githubusercontent.com/15848350/51422224-18d15e80-1bd1-11e9-96f8-9d13896cb034.gif)
this is really infuriating. Please fix it.

- YPD doesn't tell the user what resolution of the video is it downloading, add a function to let users know what resolution they are downloading.

- Another request is to enhance the UI for the ease of use. add all the setting of YPD on the very first page instead of hiding them in sidebars or show them via pressing a button it takes too much time honestly I don't like this, there are only a few settings to play with and every time I need to enable/disable a setting I need to find that setting. do something so I don't need to find a single setting 

- Make an updateable portable version if a user tries to update it your app will download a setup instead of updating the portable version files. if I need a setup to update it defeats the whole purpose.
 
**So in short, these are things you need to worry about.**
- [x] Fix the conversion process & subtitle problem in the new version.

- [x] Fix the UI for better UX & for ease of use.
- [ ] Download counter isn't showing the correct speeds.

- [ ] Add a function to let the users know what resolution of the video they are downloading.

- [ ] Updateable portable version.

**Comment by shaked6540** [comment]
I will try to fix the conversion bug if I have time this weekend 


The program should still work normal, and if you see it converting 0 files it should be done. The files should be in the folder you set it to be at


I don't know how to solve the download speed indicator problem, I don't have time to sit and think it out either, I'm in the army now and I honestly can't make time for that right now. I will try to find time to fix everything but I don't promise anything

**Comment by Droyk** [comment]
> I don't know how to solve the download speed indicator problem, I don't have time to sit and think it out either

To be honest I don't really mind it because what I want from a downloading app that it should download files at full speed that's what YPD doing so if you can't solve this problem either you can remove it or instead of showing speed, show time. simple.

>I will try to find time to fix everything but I don't promise anything

No problem, Fix it only when you can extract some time from your busy life.

**Comment by shaked6540** [comment]
A new version has been released, it hopefully fixed all the bugs (not including the download speed indicator) 

I didn't make an updateable portable version yet
And about the video resolution, what do you mean let the user know what resolution is being downloaded? the user chooses it in the settings menu

**Comment by Droyk** [comment]
Got the new update, yup fixed the bug.
now if you can add all the setting on the very first page instead of hiding them on different pages it would be really time-saving. 

>And about the video resolution, what do you mean let the user know what resolution is being downloaded? the user chooses it in the settings menu

for e.g., I always choose the highest option available
![image](https://user-images.githubusercontent.com/15848350/51805948-b693f000-2299-11e9-89fb-8b8b269207b5.png)
but you know many videos don't offer the preferred resolution so it captures the uploaded video's resolution instead. at that time user can't really know from YPD what resolution it's downloading? user's need to open the video URL to check what's the highest resolution available that's not really time saving & efficient way to check the resolution is it?

So what I want from YPD is to show the user how many types of different resolution it can provide to the user or what is the highest resolution available for the current URL.

**Comment by shaked6540** [comment]
I'm going to close this issue because I can't seem to fix the download speed indicator, as far as I see it works fine for me and I can't replicate the issues, I tried using different algorithms to calculate it but they all return the same results, at least for me

An updateable portable version would require a lot of work from my side and I simply don't have the time for it, so it's not going to happen soon, sorry 

The video resolution option is in a bit of a grey area for me because it would just clog up the download page and I don't like the idea of it too much, it's not going to come any time soon unless I change my mind, sorry for that too 

Hope you will still enjoy YPD even without these features

**Comment by Droyk** [comment]
I can understand, atleast you tried.

>Hope you will still enjoy YPD even without these features

I will.
